x-pack/plugins/ml/public/data_frame/common/pivot_group_by.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dictionary } from '../../../common/types/common';
+
+export enum PIVOT_SUPPORTED_GROUP_BY_AGGS {
+  DATE_HISTOGRAM = 'date_histogram',
+  HISTOGRAM = 'histogram',
+  TERMS = 'terms',
+}
+
+type FieldName = string;
+
+interface GroupByConfigBase {
+  field: FieldName;
+  formRowLabel: string;
+}
+
+// Don't allow an interval of '0', but allow a float interval of '0.1' with a leading zero.
+export const histogramIntervalFormatRegex = /^([1-9][0-9]*((\.)([0-9]+))?|([0](\.)([0-9]+)))$/;
+// Don't allow intervals of '0', don't allow floating intervals.
+export const dateHistogramIntervalFormatRegex = /^[1-9][0-9]*(ms|s|m|h|d|w|M|q|y)$/;
```

Comment:
We need to be careful not to accept multiple units of the larger intervals (from `w` upwards I believe`), as these are not valid Elasticsearch intervals. If you put in `3w` for example you see an error from the Pivot Preview:

![image](https://user-images.githubusercontent.com/7405507/56803041-81a42480-6819-11e9-82c4-9cfd2df30f74.png)

Reply:
I extended the interval checking to only allow a value of `1` for `w/M/y` with test coverage in 0b92f7b.